db/migrate/20200912032913_add_ancestry_to_resource_pool.rb

```diff
@@ -0,0 +1,130 @@
+class AddAncestryToResourcePool < ActiveRecord::Migration[5.2]
+  class ResourcePool < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    has_many :all_relationships, :class_name => "AddAncestryToResourcePool::Relationship", :dependent => :destroy, :as => :resource
+    include ActiveRecord::IdRegions
+  end
+
+  class Relationship < ActiveRecord::Base
+    belongs_to :resource_pool, :class_name => 'AddAncestryToResourcePool::ResourcePool', :foreign_key => :resource_pool_id
+  end
+
+  def up
+    add_column :resource_pools, :ancestry, :string
+    add_index :resource_pools, :ancestry
+
+    say_with_time("set vm ancestry from existing genealogy relationship resource information") do
+      transfer_relationships_to_ancestry(ResourcePool, 'ResourcePool', 'genealogy')
+    end
+
+    Relationship.where(:relationship => 'genealogy', :resource_type => 'ResourcePool', :resource_id => ResourcePool.all.select(:id)).delete_all
+  end
+
+  def down
+    say_with_time("create relationship records from vm ancestry") do
+      create_relationships_from_ancestry(ResourcePool.all, Relationship, 'ResourcePool', 'genealogy')
+    end
+
+    remove_column :resource_pools, :ancestry
+  end
+
+  # src is relationship, dest is vm
```

Comment:
in this case, dest is resource pool (copy and paste error)

maybe dest is something to the effect that it is the `model with ancestry column that will get changed`